spark/dl/src/main/scala/com/intel/analytics/bigdl/tensor/QuantTensor.scala

```diff
@@ -0,0 +1,1667 @@
+    _size = sizes
+    _stride = strides
+    this
+  }
+
+  override def resize(size1: Int): Tensor[T] = {
+    if (this.nDimension != 1 || this.size(1) != size1) {
+      resize(Array(size1))
+    } else {
+      this
+    }
+  }
+
+  override def resize(size1: Int, size2: Int): Tensor[T] = {
+    if (this.nDimension != 2 || this.size(1) != size1 || this.size(2) != size2) {
+      resize(Array(size1, size2))
+    } else {
+      this
+    }
+  }
+
+  override def resize(size1: Int, size2: Int, size3: Int): Tensor[T] = {
+    if (this.nDimension != 3 || this.size(1) != size1 || this.size(2) != size2 ||
+            this.size(3) != size3) {
+      resize(Array(size1, size2, size3))
+    } else {
+      this
+    }
+  }
+
+  override def resize(size1: Int, size2: Int, size3: Int, size4: Int): Tensor[T] = {
+    if (this.nDimension != 4 || this.size(1) != size1 || this.size(2) != size2 ||
+            this.size(3) != size3 ||
+            this.size(4) != size4) {
+      resize(Array(size1, size2, size3, size4))
+    } else {
+      this
+    }
+  }
+
+  override def resize(size1: Int, size2: Int, size3: Int, size4: Int, size5: Int): Tensor[T] = {
+    if (this.nDimension != 5 || this.size(1) != size1 || this.size(2) != size2 ||
+            this.size(3) != size3 || this.size(4) != size4 || this.size(5) != size5) {
+      resize(Array(size1, size2, size3, size4, size5))
+    } else {
+      this
+    }
+  }
+
+  /**
+   * Element number
+   *
+   * @return element number
+   */
+  override def nElement(): Int = {
+    if (this.nDimension == 0) {
+      0
+    } else {
+      var n = 1
+      var d = 0
+      while (d < this.nDimension) {
+        n = n * this._size(d)
+        d += 1
+      }
+      n
+    }
+  }
+
+  /**
+   * Remove the dim-th dimension and return the subset part. For instance
+   * tensor =
+   * 1 2 3
+   * 4 5 6
+   * tensor.select(1, 1) is [1 2 3]
+   * tensor.select(1, 2) is [4 5 6]
+   * tensor.select(2, 3) is [3 6]
+   *
+   * @param dim
+   * @param index
+   * @return
+   */
+  override def select(dim: Int, index: Int): Tensor[T] =
+    throw new UnsupportedOperationException(errorString)
+
+  /**
+   * Get the storage
+   *
+   * @return storage
+   */
+  override def storage(): Storage[T] =
+    throw new UnsupportedOperationException(errorString)
+
+  /**
+   * tensor offset on the storage
+   *
+   * @return storage offset, count from 1
+   */
+  override def storageOffset(): Int =
+    throw new UnsupportedOperationException(errorString)
+
+  /**
+   * The Tensor is now going to "view" the same storage as the given tensor. As the result,
+   * any modification in the elements of the Tensor will have an impact on the elements of the
+   * given tensor, and vice-versa. This is an efficient method, as there is no memory copy!
+   *
+   * @param other the given tensor
+   * @return current tensor
+   */
+  override def set(other: Tensor[T]): Tensor[T] = {
```

Comment:
Why do we need this?

Reply:
In rnn/broadcast, we need set it to empty then set it back. So add this method.